packages/java-edition/src/index.ts

```diff
@@ -44,7 +44,7 @@ export const initialize: core.ProjectInitializer = async (ctx) => {
 		const searched = new Set<string>()
 		for (let depth = 0; depth <= 2; depth += 1) {
 			for (const projectRoot of projectRoots) {
-				const files = await externals.fs.getAllFiles(projectRoot, depth + 1)
+				const files = await core.fileUtil.getAllFiles(externals, projectRoot, depth + 1)
```

Comment:
The `getAllFiles` function here is called multiple times, each time increasing its depth, therefore repeating the directory walk. It may be possible to optimize this in the future.

Reply:
I think this wouldn't be necessary if we make `getAllFiles()` breadth-first instead of depth-first, so that the earliest `pack.mcmeta` file would be the shallowest, and we can limit depth to `2` directly.

Reply:
The point would also be to prevent checking depth 2 if there is a valid pack.mcmeta at depth 1

Reply:
Hmm. It could be made into a breadth-first generator?

Reply:
But imo that isn't necessary to be done now. This will most likely already be more efficient than globby. So feel free to merge if you are fine with this.

Reply:
Yeah. I also think if it were implemented it wouldn't necessarily be an optimization. We'd just be trading more memory consumption for less IO calls.